legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/validator/MappingValidator.java

```diff
@@ -43,6 +46,8 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static org.finos.legend.pure.generated.core_relational_relational_relationalMappingExecution.Root_meta_relational_mapping_assertCompatibleTypesInRelationalPropertyMapping_RelationalPropertyMapping_$0_1$__Boolean_1_;
```

Comment:
I think this file is wrong place to add this logic as it is relational in nature. That's why you need to chnage pom too. Add it here: https://github.com/finos/legend-engine/blob/b67a48192bb4ca0d8fc26131b642d28326ac781c/legend-engine-xt-relationalStore-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperRelationalBuilder.java#L934